templates/vue/src/components/modals/NodeModal/forms/ContentForm/DragAndDropForm/index.vue

```diff
@@ -0,0 +1,203 @@
+    },
+  },
+  data() {
+    return {
+      fromBuckets: [
+        {
+          id: 1,
+          value: "",
+          itemArray: [
+            {
+              id: 1,
+              color: "#808080",
+              imageurl: "",
+              text: "",
+            },
+          ],
+        },
+      ],
+      useImages: false,
+      hideText: false,
+      nextFromBucketId: 2,
+      nextFromBucketItemId: 2,
+      nextFromBucketValue: "",
+      toBuckets: [
+        {
+          id: 200,
+          value: "",
+          itemArray: [],
+        },
+      ],
+      nextToBucketId: 201,
+      nextToBucketValue: "",
+    }
+  },
+  computed: {
+    removeButtonFromDisabled() {
+      return this.fromBuckets.length === 1
+    },
+    removeButtonToDisabled() {
+      return this.toBuckets.length === 1
+    },
+  },
+  watch: {
+    fromBuckets(newFromBuckets) {
+      this.question.answerTypes.dragDrop.fromBucketArray = newFromBuckets
+    },
+    toBuckets(newToBuckets) {
+      this.question.answerTypes.dragDrop.toBucketArray = newToBuckets
+    },
+    nextFromBucketItemId(newNextFromBucketItemId) {
+      this.question.answerTypes.dragDrop.nextFromBucketItemId = newNextFromBucketItemId
+    },
+    nextFromBucketId(newNextFromBucketId) {
+      this.question.answerTypes.dragDrop.nextFromBucketId = newNextFromBucketId
+    },
+    nextToBucketId(newNextToBucketId) {
+      this.question.answerTypes.dragDrop.nextToBucketId = newNextToBucketId
+    },
+    useImages(newUseImages) {
+      this.question.answerTypes.dragDrop.useImages = newUseImages
+    },
+    hideText(newHideText) {
+      this.question.answerTypes.dragDrop.hideText = newHideText
+    },
+  },
+  created() {
+    if (
+      !this.question.answerTypes.dragDrop.hasOwnProperty("fromBucketArray") ||
+      !this.question.answerTypes.dragDrop.hasOwnProperty("toBucketArray")
+    ) {
+      this.question.answerTypes.dragDrop.fromBucketArray = this.fromBuckets
+      this.question.answerTypes.dragDrop.toBucketArray = this.toBuckets
+      this.question.answerTypes.dragDrop.nextFromBucketItemId = this.nextFromBucketItemId
+      this.question.answerTypes.dragDrop.nextFromBucketId = this.nextFromBucketId
+      this.question.answerTypes.dragDrop.nextToBucketId = this.nextToBucketId
+      this.question.answerTypes.dragDrop.useImages = this.useImages
+      this.question.answerTypes.dragDrop.hideText = this.hideText
+    } else {
+      this.fromBuckets = this.question.answerTypes.dragDrop.fromBucketArray
+      this.toBuckets = this.question.answerTypes.dragDrop.toBucketArray
+      this.nextFromBucketItemId = this.question.answerTypes.dragDrop.nextFromBucketItemId
+      this.nextFromBucketId = this.question.answerTypes.dragDrop.nextFromBucketId
+      this.nextToBucketId = this.question.answerTypes.dragDrop.nextToBucketId
+      this.useImages = this.question.answerTypes.dragDrop.useImages
+      this.hideText = this.question.answerTypes.dragDrop.hideText
+    }
```

Comment:
Again, I wonder if the code structure can be improved so we wouldn't need to do all of this... seems tedious